src/Extractor.php

```diff
@@ -95,11 +96,18 @@ public function extract(): void
 
             // Process the previous page, while waiting for the next page ^^^^
             if ($result) {
+                $this->pageCount++;
                 foreach ($result->getEntities() as &$entity) {
                     $csvWriter->writeItem($entity);
                     $this->rowsCount++;
+
+                    // In the QueryFactory is "$top" set to the limit from the config
+                    // But "$top" is limit for one request/page, therefore we must check limit in the code
+                    if ($limit && $this->rowsCount >= $limit) {
```

Comment:
V `QueryFactory` sa nastavuje do `Query` `top`. ... no to je limit iba vramci jedneho requestu/page.
Preto sa to tu musi tiez kontrolovat. .... mozem nastavit limit/top `10000`, ... a API mi vrati 10 pages s `1000` polozkami. .... mozem nastavit limit/top na `8` a API mi vrati 1 page s 8 polozkami.